ui/plugin-system/src/model/plugins.ts

```diff
@@ -43,12 +44,17 @@ export interface PluginMetadata {
 }
 
 /**
- * All supported plugin types.
+ * All supported plugin types. A plugin's implementation must extend from `Plugin<UnknownSpec>` to be considered a valid
+ * `PluginType`.
  */
-export type PluginType = keyof SupportedPlugins;
+export type PluginType = {
```

Comment:
Your comment, _"I also realized that this basically means that consumers can now extend the plugin system with their own plugin types"_ sounds really cool. 

So in the hypothetical, `kind: 'MonitorTooltip'` or `kind: 'AlertTooltip'`, plugin use case, what would potentially live in our internal codebase vs. in Perses?

Reply:
Yeah, basically a plugin could decide "I want people to be able to write plugins for this part of my functionality" and could then define a new plugin type. So in your example, this might be a plugin (like `TimeSeriesChart`) in some other package (like `@perses-dev/panels-plugin`) deciding to make tooltips a plugin without modifying the core plugin system package.

(Note: I think for that _specific_ example we might debate whether to make those plugins part of the core plugin system since they're used by our core panel plugins, but we'd have the option to not do it.)